src/NServiceBus.Core/Serializers/XML/Serializer.cs

```diff
@@ -62,14 +57,28 @@ string GetNamespace(object message)
             return mapper.GetMappedTypeFor(message.GetType()).Namespace;
         }
 
-        StringBuilder SerializeMessage(object message)
+        string SerializeMessage(object message)
         {
             var messageBuilder = new StringBuilder();
+
+            messageBuilder.AppendLine("<?xml version=\"1.0\" ?>");
+
             var t = mapper.GetMappedTypeFor(message.GetType());
+            var baseTypes = GetBaseTypes(message);
+
+            var objectBuilder = new StringBuilder();
+            var elementName = t.SerializationFriendlyName();
+
+            WriteObject(elementName, t, message, objectBuilder);
+
+            var startElementBuilder = new StringBuilder();
+            var ns = GetNamespace(message);
+            WriteElementWithNamespaces(ns, baseTypes, startElementBuilder, elementName);
 
-            WriteObject(t.SerializationFriendlyName(), t, message, messageBuilder, true);
+            objectBuilder.Replace(string.Format("<{0}>", elementName), startElementBuilder.ToString());
```

Comment:
Not sure I like the replace thing. I mean, arguably it is hard to follow the flow of code if the flow is based on the fact that we insert some value into a string for the purpose of replacing it later. How about we introduce a specialized `Write` type of method for the purpose of writing a top-level element:

```
//Pseudocode
void WriteTopLevelObject(string name, Type type, object value, StringBuilder builder)
{
    Write(builder, type, value);
    var @namespace = InitializeNamespaces(value);
    var baseTypes = GetBaseTypes(value);
    var startElement = CreateStartElementWithNamespaces(@namespace, baseTypes, builder, name);
    builder.Insert(0, "<{0}>\n", startElement);
    builder.AppendFormat("</{0}>\n", name);
}
```

I hoped to also get rid of the insert somehow because that's not super efficient. Maybe by returning `IEnumerable<string>` from this `WriteTopLevelObject` ?